### PR TITLE
Update case gather with additional information

### PR DESCRIPTION
**Pull request by AriHrannar**
## Status
**Ready**

## Description
Gather some additional data from althingi.is and add that to cases. Like the link to althingi.is, the case status, who put it forward and what subjects are related to it.

Added status, althingi_status. althingi_link and case_creators

Current case object
```javascript
{
        "id": 1,
        "name": "fjárlög 2017",
        "number": 1,
        "case_type": "Frumvarp til laga",
        "althingi_status": "Samþykkt sem lög frá Alþingi.",
        "parliament_session": {
            "session_number": 146,
            "parliament": {
                "name": "Test-Parliament",
                "id": 1
            },
            "name": "Test-Parliament-Session",
            "created": "2017-04-25T17:41:28Z",
            "modified": "2017-04-25T17:41:28Z"
        },
        "created": "2017-08-18T23:19:32.077600Z",
        "modified": "2017-08-18T23:19:32.077683Z",
        "althingi_link": "http://www.althingi.is/dba-bin/ferill.pl?ltg=146&mnr=1",
        "case_creators": [
            {
                "parliament_member": {
                    "name": "Haraldur Benediktsson",
                    "id": 25,
                    "initials": "HarB",
                    "districtNumber": 1,
                    "party": {
                        "name": "Sjálfstæðisflokkur",
                        "id": 1,
                        "website": "http://xd.is/",
                        "created": "2017-04-25T17:41:28Z",
                        "modified": "2017-04-25T17:41:28Z"
                    },
                    "district": {
                        "name": "Norðvesturkjördæmi",
                        "id": 6,
                        "abbreviation": "Norðvest.",
                        "politicians": [
                            {
                                "name": "Elsa Lára Arnardóttir",
                                "id": 16,
                                "initials": "ELA",
                                "party": {
                                    "name": "Framsóknarflokkurinn",
                                    "id": 2,
                                    "website": "https://framsokn.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            },
                            {
                                "name": "Eva Pandora Baldursdóttir",
                                "id": 17,
                                "initials": "EPB",
                                "party": {
                                    "name": "Píratar",
                                    "id": 5,
                                    "website": "https://piratar.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            },
                            {
                                "name": "Guðjón S. Brjánsson",
                                "id": 19,
                                "initials": "GBr",
                                "party": {
                                    "name": "Samfylkingin",
                                    "id": 6,
                                    "website": "http://xs.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            },
                            {
                                "name": "Gunnar Bragi Sveinsson",
                                "id": 22,
                                "initials": "GBS",
                                "party": {
                                    "name": "Framsóknarflokkurinn",
                                    "id": 2,
                                    "website": "https://framsokn.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            },
                            {
                                "name": "Haraldur Benediktsson",
                                "id": 25,
                                "initials": "HarB",
                                "party": {
                                    "name": "Sjálfstæðisflokkur",
                                    "id": 1,
                                    "website": "http://xd.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            },
                            {
                                "name": "Lilja Rafney Magnúsdóttir",
                                "id": 35,
                                "initials": "LRM",
                                "party": {
                                    "name": "Vinstri Hreyfingin - Grænt Framboð",
                                    "id": 3,
                                    "website": "http://vg.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            },
                            {
                                "name": "Teitur Björn Einarsson",
                                "id": 53,
                                "initials": "TBE",
                                "party": {
                                    "name": "Sjálfstæðisflokkur",
                                    "id": 1,
                                    "website": "http://xd.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            },
                            {
                                "name": "Þórdís Kolbrun R. Gylfadóttir",
                                "id": 61,
                                "initials": "ÞórdG",
                                "party": {
                                    "name": "Sjálfstæðisflokkur",
                                    "id": 1,
                                    "website": "http://xd.is/",
                                    "created": "2017-04-25T17:41:28Z",
                                    "modified": "2017-04-25T17:41:28Z"
                                },
                                "promises": []
                            }
                        ],
                        "created": "2017-04-25T17:41:28Z",
                        "modified": "2017-04-25T17:41:28Z"
                    }
                }
            }
        ]
    },
```
(bit awkward with the current nested serialization and even crashes chrome for me :) )

## Todos
- [x] Structure code a bit better
- [x] Dont use static parliament session id, use latest (or latest two just in case)
- [x] Do we care about subject hierarchy?
- [x] Remove Bills model?
- [x] Should we have questions? (I think thats cases with type "q")
- [x] Tests
- [x] Why does it sometimes fail silently!!!

### CODE REVIEW

File: case_gather/cron_jobs.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 CRONLOGGER = logging.getLogger('cronJobs')
 
 class GatherCases(CronJobBase):
-    RUN_EVERY_MINS = 30
+    RUN_EVERY_MINS = 1
```
Comment (skabbi) [inline case_gather/cron_jobs.py]: I think this value seems would benefit by being configurable, especially if it and promises/cronjobs.py should share the same value. Could it be added to a settings file?

Comment (AriHrannar) [inline case_gather/cron_jobs.py]: Im working on dockerizing this project and ill be looking into the cronjobs very soon (adding gunicorn now, then working on cronjobs).

I was thinking of having that as an environmental variable, so we can just define it in our docker setup :)

Comment (skabbi) [inline case_gather/cron_jobs.py]: Sounds good. I recommend making sure all environmental variables are documented as they are easily forgotten about.

File: src/case_gather/xml_parser.py [hunk 2]
```diff
@@ -1,181 +1,138 @@
...
-        return soup
-    finally:
-        pass
+def getCaseCreatorNames(soup):
+    """
+        Takes in a soup and finds the 'framsögumaður'
+        goes through them all, adds their names to a list
+        and returns said list
+    """
+    case_creators_elements = soup.find_all("framsögumaður")
```
Comment (skabbi) [inline src/case_gather/xml_parser.py]: Why is "framsögumaður" being used here?

File: src/case_gather/services.py [hunk 3]
```diff
@@ -1,55 +1,149 @@
...
+        else:
+            sentryLogger.logToSentry('Parliament member not found: ' + case_creator_name)
+
+def create_althingi_status_to_status_map():
+    """
+     Creates a map so that we can easily retrieve a more readable status from the
+     althingi_status. This is done as the althingi status is too complicated.
+     Key and Value are stored in database
+     Map will look something like:
+     {
```
Comment (Ragnar-H) [inline src/case_gather/services.py]: We need to be careful to check the date of this comment before trusting it too much.

You know, 6 months from now when we need to touch this code again

File: src/case_gather/xml_parser.py [hunk 4]
```diff
@@ -1,181 +1,138 @@
...
-    case_numbers_gen = xml_helper.get_attribute_value(
-        soup, "mál", "málsnúmer")
+def getCaseName(soup):
+    """
+        Takes in a soup and finds the case name
+    """
+    return soup.find("málsheiti").string
 
-    xml_logger.info('have the case numbers generator')
+def getAlthingiLink(soup):
```
Comment (Ragnar-H) [inline src/case_gather/xml_parser.py]: \o/

I'm a fan of these "mini" functions
Really clarifies what one is looking for and why

File: src/case_gather/xml_parser.py [hunk 5]
```diff
@@ -1,181 +1,138 @@
...
-    finally:
-        pass
+def getCaseCreatorNames(soup):
+    """
+        Takes in a soup and finds the 'framsögumaður'
+        goes through them all, adds their names to a list
+        and returns said list
+    """
+    case_creators_elements = soup.find_all("framsögumaður")
+    #print(case_creators_elements)
```
Comment (Ragnar-H) [inline src/case_gather/xml_parser.py]: _shame_

**Comment by Ragnar-H** [comment]
I was too eager to approve the PR.

Some tests are failing :(

```
..F..........................
======================================================================
FAIL: test_get_case_creators (case_gather.xml_parser_test.GetCaseCreatorNames)
----------------------------------------------------------------------
Traceback (most recent call last):
File "/code/src/case_gather/xml_parser_test.py", line 28, in test_get_case_creators
self.assertEqual(expected, case_creators)
AssertionError: Lists differ: ['John', 'James', 'Jonathan'] != []
First list contains 3 additional elements.
First extra element 0:
'John'
- ['John', 'James', 'Jonathan']
+ []
```

File: src/case_gather/xml_parser.py (hunk 4 above)
Comment (AriHrannar) [inline src/case_gather/xml_parser.py]: Precisely :)

File: src/case_gather/xml_parser.py (hunk 2 above)
Comment (AriHrannar) [inline src/case_gather/xml_parser.py]: Its just a element in the XML from althingi (search for it [here](http://www.althingi.is/altext/xml/thingmalalisti/thingmal/?lthing=146&malnr=413))

As far as I understand, it means whoever presents the case on althingi (we save it as case creator)

**Pull request merged**